### DISCUSSION TITLE
Minification not working properly

### DISCUSSION THREAD

**Discussion by suffle** [vercel/next.js#17227] [Help] [answered]
# Bug report

When importing a library on a page (or in another file that is used on a page), the minification does not work correctly and bloats the js files.

## To Reproduce

I added a very simple test.js to my pages folder that looks like this:

```javascript
import React from "react";

import mdx from "@mdx-js/mdx";

const TestPage = props => {
  return <h1>This is a headline</h1>;
};

export default TestPage;
```
and also added some lines to my next.config.js to make the build work:

```javascript
module.exports = {
  webpack(config, { isServer }) {
    if (!isServer) {
      config.node = {
        fs: "empty",
      };
    }
    return config;
};
```
## Expected behavior

The imported mdx-Library is not used so I would expect it to be removed on minification. As you can see in the screenshot, there is a huge size dfference between the test-page with and without the import.

## Screenshots

Without the import statement
![withoutImport](https://user-images.githubusercontent.com/10942229/93672381-b4a02580-faaa-11ea-93cf-59e37efdff18.png)

With the import statement
![withImport](https://user-images.githubusercontent.com/10942229/93672387-c1bd1480-faaa-11ea-84c4-b57623970fd0.png)

## System information

- OS: Linux
- Version of Next.js: 9.5.3, and also the latest 9.5.4-canary
- Version of Node.js:12.18.3

**Comment by timneutkens** [comment, marked as answer]
MDX is probably not marked as side-effect free. This is not a Next.js bug so I'll convert it to a discussion.
